wazo_confd/sync_db.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2020 The Wazo Authors  (see the AUTHORS file)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import argparse
+import logging
+
+from xivo.chain_map import ChainMap
+from xivo.config_helper import read_config_file_hierarchy
+from xivo_dao import init_db_from_config
+from xivo_dao.alchemy.tenant import Tenant
+from xivo_dao.helpers.db_utils import session_scope
+from xivo_dao.resources.user import dao as user_dao
+from wazo_auth_client import Client as AuthClient
+
+from wazo_confd.config import DEFAULT_CONFIG, _load_key_file
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger('wazo-confd-sync-db')
+
+
+def parse_cli_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '-d', '--debug', action='store_true', help="Log debug messages",
+    )
+    parser.add_argument(
+        '-q', '--quiet', action='store_true', help='Only print warnings and errors',
+    )
+    parsed_args = parser.parse_args()
+    result = {}
+    if parsed_args.quiet:
+        logger.setLevel(logging.WARNING)
+    elif parsed_args.debug:
+        logger.setLevel(logging.DEBUG)
+    return result
+
+
+def load_config():
+    file_config = read_config_file_hierarchy(ChainMap(DEFAULT_CONFIG))
+    service_key = _load_key_file(ChainMap(file_config, DEFAULT_CONFIG))
+    return ChainMap(service_key, file_config, DEFAULT_CONFIG)
+
+
+def main():
+    parse_cli_args()
+    config = load_config()
+
+    token = AuthClient(**config['auth']).token.new(expiration=300)['token']
+
+    del config['auth']['username']
+    del config['auth']['password']
+    tenants = AuthClient(token=token, **config['auth']).tenants.list()['items']
+    auth_tenants = set(tenant['uuid'] for tenant in tenants)
+    logger.debug('wazo-auth tenants: %s', auth_tenants)
+
+    init_db_from_config(config)
+    with session_scope() as session:
+        tenants = session.query(Tenant).all()
+        confd_tenants = set(tenant.uuid for tenant in tenants)
+        logger.debug('wazo-confd tenants: %s', confd_tenants)
+
+        removed_tenants = confd_tenants - auth_tenants
+        for tenant in removed_tenants:
+            logger.info('Removing tenant: %s... (SKIP)', tenant)
```

Comment:
Why did you log `(SKIP)`?

Reply:
Because the tenant is not removed, only some resources in this tenant
But there is probably a better way to explain that than SKIP

Reply:
you're right, I did not notice that the tenant uuid was still in the confd tenant table after the cleanup. Skip makes sense then.